apps/prairielearn/src/sprocs/array_product.sql

```diff
@@ -5,7 +5,17 @@ CREATE FUNCTION
         OUT product DOUBLE PRECISION[]
     )
 AS $$
+DECLARE
+    i integer;
+    j integer := 1;
 BEGIN
-    SELECT array_agg(xi * yi) INTO product FROM unnest(x, y) as tmp (xi,yi);
+    FOR i in 1..coalesce(cardinality(x), 0) LOOP
+        IF x[i] IS NULL THEN
+            product[i] := NULL;
+        ELSE
+            product[i] := x[i] * y[j];
+            j := j + 1;
+        END IF;
+    END LOOP;
```

Comment:
I'm on board with this change but can we change the name of this function to something like `scores_to_points()`? It's no longer really doing a simple `array_product`. It's only used in one place so renaming should be safe.

Reply:
Done. Also renamed the parameters for clarity.

Reply:
Looks good!